Esri/arcgis-maps-sdk-swift-samples
# [New] Set feature request mode

## Description

This PR implements `Set feature request mode` in `Search and Query` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-SetFeatureRequestMode/Shared/Samples/Set%20feature%20request%20mode)

## Linked Issue(s)

- `swift/issues/4906`

## How To Test

- Use the picker to change the feature request mode.
- When "Manual Cache" is selected, tap "Populate" to load the feature within the screen extent.
- Pan and zoom see how the features update in the different modes.

## Screenshots

![set-feature-request-mode](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/5a96e1eb-54ff-49d8-8765-ddcb571f432b)
